lib/node_tar.js

```diff
@@ -5,36 +5,38 @@ let os = require('os')
 
 module.exports = {
 
-  call: function (cwd, file, callback) {
-    let archive = archiver('tar', { gzip: true })
+  call: function (cwd, file) {
+    return new Promise(function (resolve, reject) {
+      let archive = archiver('tar', { gzip: true })
 
-    let ig = ignore()
-    if (fs.existsSync('.gitignore')) {
-      ig = ig.add(fs.readFileSync('.gitignore').toString())
-    }
-    let filter = ig.createFilter()
-
-    archive.on('finish', function (err) {
-      if (err) { throw err }
-      callback()
-    })
+      let ig = ignore()
+      if (fs.existsSync('.gitignore')) {
+        ig = ig.add(fs.readFileSync('.gitignore').toString())
+      }
+      let filter = ig.createFilter()
 
-    let output = fs.createWriteStream(file)
-    archive.pipe(output)
-    var data = {}
-    if (os.platform() === 'win32') {
-      data.mode = 0o0755
-    }
+      archive.on('finish', function (err) {
+        if (err) { reject(err) }
+        resolve()
+      })
 
-    archive.bulk([
-      { expand: true,
-        cwd: cwd,
-        src: ['**'],
-        dest: false,
-        filter: filter,
-        data: data,
-        dot: true
+      let output = fs.createWriteStream(file)
+      archive.pipe(output)
+      var data = {}
+      if (os.platform() === 'win32') {
```

Comment:
we've found that sometimes node returns `windows` here, so you'll want to check that as well as `win32`